### PR TITLE
fdroidserver: add new features

### PR DESCRIPTION
**Pull request by eighthave**
Some new features are optionally enabled by including the biplist and pycountry libs, and _rclone_ util.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/f/fdroidserver.rb
Comment (chenrui333) [inline Formula/f/fdroidserver.rb:1]: might also good to add `revision 1` on line 9

File: Formula/f/fdroidserver.rb [hunk 1]
```diff
@@ -385,10 +398,20 @@ def caveats
...
           name: Development
+          icon: category_development.png
         System:
           name: System
+          icon: category_system.png
       YAML
 
+      # fake PNGs, they just start with PNG file magic bytes
+      (testpath/"fdroid/config/category_development.png").write <<~PNG
+        \x89\x50\x4E\x47\x0D\x0A\x1A\x0A\x00\x00\x00
```
Comment (p-linnane) [inline Formula/f/fdroidserver.rb:409]: This is causing `An error occurred while FormulaAudit/PythonVersions cop was inspecting /opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/f/fdroidserver.rb:1:0.`

Any ideas @Homebrew/core?

Comment (cho-m) [reply Formula/f/fdroidserver.rb:409]: Not sure on error, but if a fake PNG is needed then can use `test_fixtures("test.png")` which uses 
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/support/fixtures/test.png

Comment (p-linnane) [reply Formula/f/fdroidserver.rb]: I've updated the test to use this instead. I've also run `brew update-python-resources` which added a few more that were not previously listed as resources.

**Comment by p-linnane** [comment]
The `androguard` resource requires `frida`. We will need to (at a minimum) add a `node` build dependency.

**Comment by eighthave** [comment]
I'd prefer to use a slightly older version of _androguard_ that doesn't use _frida_.  This project is just using the APK parsing stuff, not all the features.

**Comment by p-linnane** [comment]
> I'd prefer to use a slightly older version of _androguard_ that doesn't use _frida_. This project is just using the APK parsing stuff, not all the features.

You'll need to define this within the project. Currently anyone that runs something like `pipgrip` is going to be offered the latest version of `androguard` which includes the `frida` dependency.

**Comment by eighthave** [comment]
This should be ready to go!  In the interest of getting incremental progress, I'll leave the _androguard_ update for another day.  _androguard_ 4.1.x is going fast in a new direction, so upstream is going to switch to a simpler library anyway.  

Thanks for all your help!

**Review by p-linnane — APPROVED** [review]

File: Formula/f/fdroidserver.rb [hunk 2]
```diff
@@ -74,6 +75,11 @@ class Fdroidserver < Formula
     sha256 "6765386e3ab87f569b276988742039baab087b2cdb01e809d74e74503c2faafe"
   end
 
+  resource "biplist" do
+    url "https://files.pythonhosted.org/packages/3e/56/2db170a498c9c6545cda16e93c2f2ef9302da44802787b45a8a520d01bdb/biplist-1.0.3.tar.gz"
+    sha256 "4c0549764c5fe50b28042ec21aa2e14fe1a2224e239a1dae77d9e7f3932aa4c6"
+  end
```
[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/f/fdroidserver.rb:83]: yeah, it got removed in the new version bump, https://github.com/Homebrew/homebrew-core/pull/199162

@eighthave can you file a new PR for the update?

**Pull request merged**